dataactbroker/handlers/fileHandler.py

```diff
@@ -639,28 +639,52 @@ def upload_detached_file(self, create_credentials):
             }
 
             job_data = {}
+            # Check for existing submission
+            existing_submission_id = request_params.get('existing_submission_id')
+            if existing_submission_id:
+                existing_submission = True
+                existing_submission_obj = sess.query(Submission).\
+                    filter_by(submission_id=existing_submission_id).\
+                    one()
+                jobs = sess.query(Job).filter(Job.submission_id == existing_submission_id)
+                # set all jobs to their initial status of "waiting"
+                jobs[0].job_status_id = JOB_STATUS_DICT['waiting']
+                sess.commit()
+
+            else:
+                existing_submission = None
+                existing_submission_obj = None
+
             for request_field, submission_field in request_job_mapping.items():
                 if request_field in request_params:
                     job_data[submission_field] = request_params[request_field]
                 # all of those fields are required
-                else:
+                # unless existing_submission_id exists
+                elif 'existing_submission_id' not in request_params and request_field in request_params:
                     raise ResponseException('{} is required'.format(request_field), StatusCode.CLIENT_ERROR, ValueError)
 
+            if existing_submission_obj is not None:
+                formatted_start_date = existing_submission_obj.reporting_start_date
+                formatted_end_date = existing_submission_obj.reporting_end_date
+                cgac_code = existing_submission_obj.cgac_code
+            else:
+                sub_tier_agency = sess.query(SubTierAgency).\
+                    filter_by(sub_tier_agency_code=request_params["agency_code"]).one()
+                cgac_code = sub_tier_agency.cgac.cgac_code
+                date_format = '%d/%m/%Y'
+                try:
+                    formatted_start_date = datetime.strptime(job_data['reporting_start_date'], date_format).date()
+                    formatted_end_date = datetime.strptime(job_data['reporting_end_date'], date_format).date()
+                except ValueError:
+                    raise ResponseException("Date must be provided as DD/MM/YYYY", StatusCode.CLIENT_ERROR, ValueError)
+
             # get the cgac code associated with this sub tier agency
-            sub_tier_agency = sess.query(SubTierAgency).\
-                filter_by(sub_tier_agency_code=request_params["agency_code"]).one()
-            job_data["cgac_code"] = sub_tier_agency.cgac.cgac_code
+            job_data["cgac_code"] = cgac_code
             job_data["d2_submission"] = True
 
             # convert submission start/end dates from the request into Python date objects
```

Comment:
Fix the comment? This should be above the "try" you moved, you probably don't actually even need this comment here and can just smush the lines below it up right under the d2_submission getting set